contracts/QuestFactory.sol

```diff
@@ -7,16 +7,20 @@ import {Erc1155Quest} from './Erc1155Quest.sol';
 import {RabbitHoleReceipt} from './RabbitHoleReceipt.sol';
 import {OwnableUpgradeable} from '@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol';
 import '@openzeppelin/contracts-upgradeable/utils/cryptography/ECDSAUpgradeable.sol';
+import '@openzeppelin/contracts-upgradeable/access/AccessControlUpgradeable.sol';
 
-contract QuestFactory is Initializable, OwnableUpgradeable {
+contract QuestFactory is Initializable, OwnableUpgradeable, AccessControlUpgradeable {
     error QuestIdUsed();
     error OverMaxAllowedToMint();
     error AddressNotSigned();
     error AddressAlreadyMinted();
     error InvalidHash();
+    error InavlidRoleToCreateQuest();
 
     event QuestCreated(address indexed creator, address indexed contractAddress, string contractType);
 
+    bytes32 public constant CREATE_QUEST_ROLE = keccak256("CREATE_QUEST_ROLE");
```

Comment:
Doesn't inserting this var here mess up the storage layout?

Reply:
no because it's a constant it's not a storage variable, the compiler treats it like a hardcoded variable.